front/lib/api/assistant/legacy_agent.ts

```diff
@@ -0,0 +1,556 @@
+      }
+    : {
+        providerId: GPT_4_TURBO_MODEL_CONFIG.providerId,
+        modelId: GPT_4_TURBO_MODEL_CONFIG.modelId,
+      };
+
+  const contextSize = auth.isUpgraded()
+    ? GPT_4_TURBO_MODEL_CONFIG.contextSize
+    : GPT_3_5_TURBO_MODEL_CONFIG.contextSize;
+
+  // Turn the conversation into a digest that can be presented to the model.
+  const modelConversationRes = await renderConversationForModel({
+    conversation,
+    model,
+    prompt,
+    allowedTokenCount: contextSize - MIN_GENERATION_TOKENS,
+  });
+
+  if (modelConversationRes.isErr()) {
+    return modelConversationRes;
+  }
+
+  const config = cloneBaseConfig(
+    DustProdActionRegistry["assistant-v2-inputs-generator"].config
+  );
+  config.MODEL.function_call = specification.name;
+  config.MODEL.provider_id = model.providerId;
+  config.MODEL.model_id = model.modelId;
+
+  const res = await runActionStreamed(
+    auth,
+    "assistant-v2-inputs-generator",
+    config,
+    [
+      {
+        conversation: modelConversationRes.value.modelConversation,
+        specification,
+        prompt,
+      },
+    ]
+  );
+
+  if (res.isErr()) {
+    return new Err(
+      new Error(
+        `Error generating action inputs: [${res.error.type}] ${res.error.message}`
+      )
+    );
+  }
+
+  const { eventStream } = res.value;
+
+  const output: Record<string, string | boolean | number> = {};
+
+  for await (const event of eventStream) {
+    if (event.type === "error") {
+      return new Err(
+        new Error(`Error generating action inputs: ${event.content.message}`)
+      );
+    }
+
+    if (event.type === "block_execution") {
+      const e = event.content.execution[0][0];
+      if (e.error) {
+        return new Err(new Error(`Error generating action inputs: ${e.error}`));
+      }
+
+      if (event.content.block_name === "OUTPUT" && e.value) {
+        const v = e.value as any;
+        for (const k in v) {
+          if (
+            typeof v[k] === "string" ||
+            typeof v[k] === "boolean" ||
+            typeof v[k] === "number"
+          ) {
+            output[k] = v[k];
+          }
+        }
+      }
+    }
+  }
+
+  return new Ok(output);
+}
+
+/**
+ * Agent execution.
+ */
+
+// This interface is used to execute an agent. It is not in charge of creating the AgentMessage,
+// nor updating it (responsability of the caller based on the emitted events).
+export async function* runLegacyAgent(
+  auth: Authenticator,
+  configuration: LightAgentConfigurationType,
+  conversation: ConversationType,
+  userMessage: UserMessageType,
+  agentMessage: AgentMessageType
+): AsyncGenerator<
+  | AgentErrorEvent
+  | AgentActionEvent
+  | AgentActionSuccessEvent
+  | GenerationTokensEvent
+  | AgentGenerationSuccessEvent
+  | AgentGenerationCancelledEvent
+  | AgentMessageSuccessEvent
+  | RetrievalParamsEvent
+  | DustAppRunParamsEvent
+  | DustAppRunBlockEvent
+  | TablesQueryParamsEvent
+  | TablesQueryOutputEvent,
```

Comment:
What about moving this to a proper type so we can avoid updating so many places?

Reply:
Orthogonal to this PR but yes bullish + we'll have to find a sustainable mechanism as we refactor the way we interface actions :+1: